src/components/SettlementButton.js

```diff
@@ -40,40 +40,36 @@ const defaultProps = {
 };
 
 class SettlementButton extends React.Component {
-    constructor(props) {
-        super(props);
+    componentDidMount() {
+        PaymentMethods.openPaymentsPage();
+    }
 
+    getButtonOptionsFromProps() {
         const buttonOptions = [];
 
-        if (props.currency === CONST.CURRENCY.USD && Permissions.canUsePayWithExpensify(props.betas) && Permissions.canUseWallet(props.betas)) {
+        if (this.props.currency === CONST.CURRENCY.USD && Permissions.canUsePayWithExpensify(this.props.betas) && Permissions.canUseWallet(this.props.betas)) {
             buttonOptions.push({
-                text: props.translate('iou.settleExpensify'),
+                text: this.props.translate('iou.settleExpensify'),
                 icon: Expensicons.Wallet,
                 value: CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
             });
         }
 
-        if (props.shouldShowPaypal) {
+        if (this.props.shouldShowPaypal) {
             buttonOptions.push({
-                text: props.translate('iou.settlePaypalMe'),
+                text: this.props.translate('iou.settlePaypalMe'),
                 icon: Expensicons.PayPal,
                 value: CONST.IOU.PAYMENT_TYPE.PAYPAL_ME,
             });
         }
 
         buttonOptions.push({
-            text: props.translate('iou.settleElsewhere'),
+            text: this.props.translate('iou.settleElsewhere'),
             icon: Expensicons.Cash,
             value: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
         });
 
-        this.state = {
-            buttonOptions,
-        };
-    }
-
-    componentDidMount() {
-        PaymentMethods.openPaymentsPage();
+        return buttonOptions;
```

Comment:
@Delgee Thanks for getting this up to fast, let's avoid the unnecessary `buttonOptions` array variable and return the button options directly like this:

```    
getButtonOptionsFromProps() {
    if (this.props.currency === CONST.CURRENCY.USD && Permissions.canUsePayWithExpensify(this.props.betas) && Permissions.canUseWallet(this.props.betas)) {
        return [{
            text: this.props.translate('iou.settleExpensify'),
            icon: Expensicons.Wallet,
            value: CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
        }];
    }
    if (this.props.shouldShowPaypal) {
        return [{
            text: this.props.translate('iou.settlePaypalMe'),
            icon: Expensicons.PayPal,
            value: CONST.IOU.PAYMENT_TYPE.PAYPAL_ME,
        }];
    }
    return [{
        text: this.props.translate('iou.settleElsewhere'),
        icon: Expensicons.Cash,
        value: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
    }];
}
```

Reply:
Thanks for the review, Looks like `ButtonWithMenu` component needs options as an array. So we probably should not return it like that.
https://github.com/Expensify/App/blob/cb93661530860f65e4aae0bc7ba7fe3c8db3e643/src/components/ButtonWithMenu.js#L25-L33
https://github.com/Expensify/App/blob/cb93661530860f65e4aae0bc7ba7fe3c8db3e643/src/components/ButtonWithMenu.js#L59-L94

Reply:
@Delgee notice I'm returning the options as an array haha:

```
return [{
            text: this.props.translate('iou.settlePaypalMe'),
            icon: Expensicons.PayPal,
            value: CONST.IOU.PAYMENT_TYPE.PAYPAL_ME,
        }];
```

Reply:
@Delgee notice I'm returning the options as an array:

```
[{
  text: this.props.translate('iou.settlePaypalMe'),
  icon: Expensicons.PayPal,
  value: CONST.IOU.PAYMENT_TYPE.PAYPAL_ME,
}];
```

this is called [array literal syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#:~:text=Array%20literals,the%20number%20of%20arguments%20specified.), you can define arrays with items in them.

Reply:
@jasperhuangg  Yeah, do you think it is unnecessary to have "settleElsewhere" option if the user has Paypal? If we pass full array user will have dropdown button instead of one option.

Reply:
ahh yes, sorry, I'm realizing now that the if statements in the original code weren't early returns, my bad, what you had original looks good then, thanks for clarifying